crates/symbolicator/src/services/symbolication.rs

```diff
@@ -1500,18 +1501,18 @@ impl TempSymbolProvider {
             .ok()
     }
 
-    fn into_missing_ids(self) -> BTreeSet<DebugId> {
-        self.missing_ids.into_inner()
+    fn missing_ids(&self) -> Vec<DebugId> {
+        self.missing_ids.read().iter().copied().collect()
     }
 }
 
+#[async_trait]
 impl SymbolProvider for TempSymbolProvider {
-    fn fill_symbol(
+    async fn fill_symbol(
         &self,
-        module: &dyn Module,
-        _frame: &mut dyn minidump_processor::FrameSymbolizer,
-    ) -> Result<(), minidump_processor::FillSymbolError> {
-        // TODO(ja): Deduplicate this. Probably should use a different map key, ...
+        module: &(dyn Module + Sync),
+        _frame: &mut (dyn FrameSymbolizer + Send),
+    ) -> Result<(), FillSymbolError> {
         let debug_id = module
             .debug_identifier()
             .and_then(|id| DebugId::from_str(&id).ok())
```

Comment:
just a NOTE, and maybe an obsolete one given that we are changing these IDs upstream:
I have found through profiling that debug-id parsing is surprisingly expensive, so avoiding that might be a good idea.